tensorflow/core/kernels/fixed_length_record_reader_op.cc

```diff
@@ -62,14 +64,20 @@ class FixedLengthRecordReader : public ReaderBase {
 
   Status ReadLocked(string* key, string* value, bool* produced,
                     bool* at_end) override {
-    if (input_buffer_->Tell() >= file_pos_limit_) {
+    if (input_buffer_->Tell() >= file_pos_limit_ ||
+        input_buffer_->Tell() + record_bytes_ > file_pos_limit_) {
```

Comment:
The error case will be signaled by `ReadNBytes()`, when the reading position is out of the range of file. With the original condition, it is possible to read a record containing some bytes of footer. Is this the expected behavior?